packages/next/src/app-dir/server.ts

```diff
@@ -32,3 +38,48 @@ export function experimental_createTRPCNextAppDirServer<
     return (client[procedureType] as any)(fullPath, ...callOpts.args);
   }) as CreateTRPCProxyClient<TRouter>;
 }
+
+/**
+ * @internal
+ */
+export type TRPCActionHandler<TProcedure extends AnyProcedure> = ((
+  ...args: inferHandlerInput<TProcedure>
+) => inferProcedureOutput<TProcedure>) & {
+  $proc: TProcedure;
+};
+
+/**
+ * @internal
+ */
+export type AnyTRPCActionHandler = TRPCActionHandler<AnyProcedure>;
+
+export function experimental_createServerActionHandler<
+  TInstance extends AnyTRPCInstance,
+>(
+  _t: AnyTRPCInstance,
+  opts: {
+    createContext: () => MaybePromise<TInstance['_config']['$types']['ctx']>;
+  },
+) {
+  // TODO allow this to take a `TRouter` in addition to a `AnyProcedure`
```

Comment:
Why?

Reply:
Make a full router into actions would be cool

Reply:
And then you'd pass it into the hook (or just <form action={}/>) like `actions.post.create`

Yea that'd be cool :)